### ISSUE TITLE
Viewing active projects in a sector / country

### ISSUE THREAD

**Issue by stevieflow** [devinit/D-Portal#333]
The click thru to a sector, eg:

http://d-portal.org/ctrack.html?publisher=46002&tongue=eng#view=list_transactions&sector_code=31110

this lists - I think - all projects in 31220 sector code - 49 projects:

![all](https://cloud.githubusercontent.com/assets/1209944/20574990/29bbeff0-b1af-11e6-95da-9f2a72cbdd12.png)

However, what I really want is a list of the active ones.  

We have a list of the active projects for this publisher here:

http://d-portal.org/ctrack.html?publisher=46002#view=active

I tried my luck, and managed to get what I need:

http://d-portal.org/ctrack.html?publisher=46002#view=active&sector_code=31110 

![active](https://cloud.githubusercontent.com/assets/1209944/20575033/601fec68-b1af-11e6-938a-2ab3c5c27af9.png)

The request is : could it be possible to navigate to such as list, rather than hack it (slightly), somehow? And- how about an active view of the list_transactions view?

**Comment by stevieflow** [comment]
(it is great that you can do this btw!)

**Comment by notshi** [comment]
Thanks! The new Discover feature should satiate this - we're still tightening things up so the page is wip but appreciate any feedback.

From

http://d-portal.org/ctrack.html?#view=search

I narrowed the search to get

http://d-portal.org/ctrack.html?search&sector_code=31110&publisher=46002#view=main

which should have all the bits you're looking for under those filters; ie. active projects, etc.

**Comment by stevieflow** [comment]
Wow - this is really useful!

One request - could the active/closed be an actual search filer on http://d-portal.org/ctrack.html?#view=search perhaps ?  Reason for this, is that we then get to a screen of active projects with the table/graphs/map etc - otherwise we just get a list:

http://d-portal.org/ctrack.html?search&country=TZ&sector_group=311#view=donor_activities&funder=AfDB

**Comment by Mattdevinit** [comment]
thanks @stevieflow - great idea (and glad this is proving useful).  Will check out how/when we can add that as a filter (prob won't be for when it goes live on homepage next week, but will try and prioritise)

**Comment by stevieflow** [comment]
Can I just check - more widely - how d-portal processes the sector codes - especially between the three and five digit codes

So: 

Does:
http://d-portal.org/ctrack.html?search&sector_group=311#view=main

include any sub 311 sector codes, such as 31191 - or are these separate ?

I cant seem to find anything via a five digit code directly:

http://d-portal.org/ctrack.html?search&sector_group=31140#view=main
http://d-portal.org/ctrack.html?search&sector_group=31110#view=main

**Comment by notshi** [comment]
Could be that sector_group and sector_code are two different elements?

Using sector_code for the 5 digit numbers will get you these results. Is this ok?

http://d-portal.org/ctrack.html?search&sector_code=31140#view=main

**Comment by stevieflow** [comment]
Hi @shi - ah ok - that's really useful

So - from the main page (http://d-portal.org/ctrack.html?#view=search) - could sector code be an option? Might get confusing with the category too though...?



(sorry , this thread is going off topic - but all very good and useful, if you don't mind)

**Comment by stevieflow** [comment]
PS: does the sector group *include* the sector codes?

**Comment by stevieflow** [comment]
Other suggestion....

Take a look at http://d-portal.org/ctrack.html?search&publisher=46002&sector_code=31161#view=main

The "Where does the money go?" graph isnt terribly useful (but still good confirmation)

The one piece of information I'd like (that I've seen elsewhere) here is the countries that are involved in these grants.  Rather than get wrapped up in lots of conditionals, it might be an idea to include a country pie chart in these views, as you do in the publisher pages: http://d-portal.org/ctrack.html?publisher=46002&tongue=eng#view=publisher ?

**Comment by Mattdevinit** [comment]
Yep that's a good call, to show the recipient pie charts rather than the publisher ones, for search results.  But then you have same issue reversed if you are searching for a recipient country (you then want to see the pie showing where the money comes from, not which country it goes to)?

**Comment by Mattdevinit** [comment]
Interested to know too if you think there's any value (when seeing the Search results) of having the stats (under 'Currently published in IATI' which shows total projects, active projects etc) above the pie charts -eg at the top)?

**Comment by stevieflow** [comment]
@Mattdevinit thanks - so:

- pie charts - I wouldn't try to be complicated / conditional and just show *all* pie charts for the results - money from; sectors; countries.  As mentioned, it's kind of useful to see a "full pie" sometimes, to confirm what you are looking at / have selected imho.  It may also be easier for @shi and @xriss to implement the same view for all search results...

- I do find value in the project stats boxes - have been using that for a filter for active.  Even it that were addressed via https://github.com/devinit/D-Portal/issues/333#issuecomment-262757645 I would still keep it

Basically, dont be clever, and dont lose anything !

**Comment by Mattdevinit** [comment]
@stevieflow sorry, maybe that was a bit unclearly phrased - was just musing on whether swapping the stats so they feature above the pie charts would be useful for users

**Comment by stevieflow** [comment]
@Mattdevinit stats above charts could be interesting / useful - maybe in "smaller" boxes or something*

* and with that, my UI/UX work is done :)

**Comment by notshi** [comment]
Thanks for the feedback, @stevieflow - definitely agree with showing 'all' the things for the results.

As it stands right now, we've not had a chance to curate the sections to display so it looks like we'll need a different page for search results depending on the search terms; ie. the results will make more sense in context and hopefully more useful.
